graphql/models/article.js

```diff
@@ -1,111 +1,121 @@
 const mongoose = require('mongoose')
 const Schema = mongoose.Schema
 
-const { computeMajorVersion, computeMinorVersion } = require('../helpers/versions.js')
+const {
+  computeMajorVersion,
+  computeMinorVersion,
+} = require('../helpers/versions.js')
 const { prefixRulesWith, sanitizeTemplate } = require('../helpers/preview.js')
 
 const ArticleContributorSchema = new Schema({
   user: {
     type: mongoose.Schema.Types.ObjectId,
-    ref: 'User'
+    ref: 'User',
   },
-  roles: [String]
+  roles: [String],
 })
 
-const articleSchema = new Schema({
-  title: {
-    type: String,
-    required: true,
-    default: 'autocreated'
-  },
-  owner: {
-    type: Schema.Types.ObjectId,
-    ref: 'User'
-  },
-  contributors: [ArticleContributorSchema],
-  zoteroLink: {
-    type: String,
-    default: ''
-  },
-  workingVersion: {
-    md: {
+const articleSchema = new Schema(
+  {
+    title: {
       type: String,
-      default: ''
-    },
-    bib: {
-      type: String,
-      default: ''
-    },
-    metadata: {
-      type: Schema.Types.Mixed,
-      default: {}
+      required: true,
+      default: 'autocreated',
     },
-  },
-  versions: [
-    {
-      type: Schema.Types.ObjectId,
-      ref: 'Version'
-    }
-  ],
-  tags: [
-    {
+    owner: {
       type: Schema.Types.ObjectId,
-      ref: 'Tag'
-    }
-  ],
-  preview: {
-    stylesheet: {
-      type: String,
-      default: '',
-      set: prefixRulesWith.bind(null, 'stylo-pagedjs-container')
+      ref: 'User',
     },
-    template: {
+    contributors: [ArticleContributorSchema],
+    zoteroLink: {
       type: String,
       default: '',
-      set: sanitizeTemplate
-    }
-  },
-  collaborativeSession: {
-    id: {
-      type: Schema.Types.ObjectId,
     },
-    creator: {
-      type: Schema.Types.ObjectId,
-      ref: 'User'
+    workingVersion: {
+      md: {
+        type: String,
+        default: '',
+      },
+      bib: {
+        type: String,
+        default: '',
+      },
+      metadata: {
+        type: Schema.Types.Mixed,
+        default: {},
+        get: (metadata) => metadata ?? {},
```

Comment:
@ggrossetie j'ai appliqué ce _getter_ sinon une erreur `Cannot destructure property 'id' of 'metadata' as it is null.` s'affichait lors des preview. T'en penses quoi ?